app/sdk/user/UserManager.java

```diff
@@ -0,0 +1,165 @@
+package sdk.user;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import play.Play;
+import play.libs.ws.WSClient;
+import play.libs.ws.WSRequest;
+import play.libs.ws.WSResponse;
+import sdk.pushNotifications.PushNotificationResponse;
+import sdk.pushNotifications.PushNotificationUtils;
+import sdk.roles.RoleManager;
+import sdk.roles.models.AppRole;
+import sdk.user.models.AppUser;
+import sdk.user.query.Clause;
+import sdk.user.query.Condition;
+import sdk.user.query.UserQuery;
+import sdk.utils.Constants;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import static sdk.utils.ValidationUtils.NullOrEmpty;
+
+public class UserManager {
+    private String programmaticUserToken;
+    private WSClient wsClient;
+    private ObjectMapper objectMapper;
```

Comment:
should this be static?

Reply:
No it's lazily loaded, so if role manager is never used it is never created.